shell/platform/embedder/embedder.cc

```diff
@@ -147,17 +136,33 @@ FlutterResult FlutterEngineRun(size_t version,
         SAFE_ACCESS(args, command_line_argv, nullptr));
   }
 
-  static std::once_flag once_shell_initialization;
-  std::call_once(once_shell_initialization, [&]() {
-    fxl::CommandLine null_command_line;
-    shell::Shell::InitStandalone(
-        std::move(command_line),
-        icu_data_path,  // icu data path default lookup.
-        ""              // application library not supported in JIT mode.
-    );
-  });
-
-  shell::PlatformViewEmbedder::DispatchTable table = {
+  blink::Settings settings = shell::SettingsFromCommandLine(command_line);
+  settings.icu_data_path = icu_data_path;
+  settings.main_dart_file_path = args->main_path;
+  settings.packages_file_path = args->packages_path;
+  settings.assets_path = args->assets_path;
+  settings.task_observer_add = [](intptr_t key, fxl::Closure callback) {
+    fml::MessageLoop::GetCurrent().AddTaskObserver(key, std::move(callback));
+  };
+  settings.task_observer_remove = [](intptr_t key) {
+    fml::MessageLoop::GetCurrent().RemoveTaskObserver(key);
+  };
+
```

Comment:
The embedder needs to check for the presence of a kernel and configure `settings` with the kernel asset paths:

Example:
```
  const char kVMKernelSnapshotFileName[] = "platform.dill";
  const char kApplicationKernelSnapshotFileName[] = "kernel_blob.bin";
  std::string platform_kernel_path =
      fml::paths::JoinPaths({settings.assets_path, kVMKernelSnapshotFileName});
  std::string application_kernel_path = fml::paths::JoinPaths(
      {settings.assets_path, kApplicationKernelSnapshotFileName});

  if (files::IsFile(platform_kernel_path) &&
      files::IsFile(application_kernel_path)) {
    settings.kernel_snapshot_path = platform_kernel_path;
    settings.application_kernel_asset = kApplicationKernelSnapshotFileName;
  }
```